static/app/components/modals/metricWidgetViewerModal.tsx

```diff
@@ -0,0 +1,319 @@
+                  value={title}
+                  placeholder={stringifyMetricWidget(metricWidgetQueryParams)}
+                  onChange={e => {
+                    setTitle?.(e.target.value ?? defaultTitle);
+                  }}
+                />
+              ) : (
+                <h3>{title}</h3>
+              )}
+              <Button
+                aria-label="Edit Title"
+                size="sm"
+                borderless
+                icon={
+                  isEditingTitle ? <IconCheckmark size="sm" /> : <IconEdit size="sm" />
+                }
+                priority={isEditingTitle ? 'primary' : 'default'}
+                onClick={() => {
+                  setIsEditingTitle(curr => !curr);
+                }}
+              />
+            </WidgetTitleRow>
+            {widget.description && (
+              <Tooltip
+                title={widget.description}
+                containerDisplayMode="grid"
+                showOnlyOnOverflow
+                isHoverable
+                position="bottom"
+              >
+                <WidgetDescription>{widget.description}</WidgetDescription>
+              </Tooltip>
+            )}
+          </WidgetHeader>
+        </Header>
+        <Body>
+          <Query
+            widget={metricWidgetQueryParams}
+            projects={selection.projects}
+            onChange={handleChange}
+            contextMenu={
+              <ContextMenu
+                metricsQuery={{
+                  mri: metricWidgetQueryParams.mri,
+                  query: metricWidgetQueryParams.query,
+                  op: metricWidgetQueryParams.op,
+                  groupBy: metricWidgetQueryParams.groupBy,
+                  projects: selection.projects,
+                  datetime: selection.datetime,
+                  environments: selection.environments,
+                }}
+              />
+            }
+          />
+          <MetricWidget
+            widget={metricWidgetQueryParams}
+            onChange={(_, data) => {
+              handleChange(data);
+            }}
+            datetime={selection.datetime}
+            projects={selection.projects}
+            environments={selection.environments}
+          />
+          <MetricDetails widget={metricWidgetQueryParams} />
+        </Body>
+        <Footer>
+          <ButtonBar gap={1}>
+            <Button
+              onClick={() => {
+                closeModal();
+                router.push(
+                  getDdmUrl(organization.slug, {
+                    widgets: [metricWidgetQueryParams],
+                    project: selection.projects,
+                    ...selection.datetime,
+                    environment: selection.environments,
+                  })
+                );
+              }}
+            >
+              {t('Open in Metrics')}
+            </Button>
+            <Button
+              onClick={() => {
+                closeModal();
+              }}
```

Comment:
Does anything prevent us from doing this?
```suggestion
              onClick={closeModal}
```

Reply:
absolutely nothing!